tests/subsys/logging/log_stack/CMakeLists.txt

```
@@ -0,0 +1,9 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(log_stack)
+
+target_sources_ifdef(CONFIG_CPLUSPLUS app PRIVATE src/maincxx.cxx)
+
+target_sources_ifndef(CONFIG_CPLUSPLUS app PRIVATE src/main.c)
```

actually, because `maincxx.cxx` and `main.c` are identical and including the same file, so a much better approach is to request CMake to compile `main.c` using the C++ compiler, like this:
```suggestion
target_sources(app PRIVATE src/main.c)
if(CONFIG_CPLUSPLUS)
  # Feel free to add a good comment here.
  set_source_files_properties(src/main.c PROPERTIES LANGUAGE CXX)
endif()
```

and then remove `maincxx.cxx` as well as `test.inc`.
Have just a single `main.c` file.

Cleaning of other places using `test.inc` can be done in seperate PR.